Userland/Libraries/LibWeb/Layout/BlockContainer.cpp

```diff
@@ -12,11 +12,31 @@ namespace Web::Layout {
 BlockContainer::BlockContainer(DOM::Document& document, DOM::Node* node, NonnullRefPtr<CSS::StyleProperties> style)
     : Box(document, node, move(style))
 {
+    if (dom_node()) {
+        StringBuilder builder;
+        builder.append(dom_node()->class_name());
+        if (MUST(builder.to_string()) == "HTMLInputElement") {
```

Comment:
This can be:
```cpp
if (dom_node()->is_html_input_element()) {
```

Reply:
I did something similar elsewhere, but it took the form `if (foo.dom_node() && foo.dom_node()->is_html_input_element())`.  Is this okay?